### ISSUE TITLE
[a11y]: TileGroup group label announced incorrectly

### ISSUE THREAD

**Issue by manudeep9798** [carbon-design-system/carbon#22358]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.84.0

### React version

19.X

### Automated testing tool and ruleset

QA

### Assistive technology

Voice over

### Description

When focus enters a tile group that contains a legend with an info icon, the group label is announced while focus is on the info icon. However, when focus moves to the actual tiles within the group, the group label is no longer announced by the screen reader.

Group label should be announced when the focus first enters the tiles not in the legend text

### Reproduction/example

https://stackblitz.com/edit/vitejs-vite-3xsjijqp?file=src%2FApp.tsx

### Steps to reproduce

1. Go to the demo: https://stackblitz.com/edit/vitejs-vite-3xsjijqp?file=src%2FApp.tsx
2. Navigate to the info icon within the "Test Name" tile group legend. Observe that the screen reader announces the group label.
3. Continue navigating to the tiles within the same tile group.
4. Observe that the group label is no longer announced when focus moves to the tiles.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
